deploy-cloudrun/src/cloudRun.ts

```diff
@@ -0,0 +1,258 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as core from '@actions/core';
+import { GaxiosResponse } from 'gaxios';
+import { google, run_v1 } from 'googleapis';
+import {
+  GoogleAuth,
+  JWT,
+  Compute,
+  UserRefreshClient,
+} from 'google-auth-library';
+import { Service } from './service';
+
+/**
+ * Available options to create the client.
+ *
+ * @param credentials GCP JSON credentials (default uses ADC).
+ * @param endpoint GCP endpoint (useful for testing).
+ */
+type ClientOptions = {
+  credentials?: string;
+  projectId?: string;
+};
+
+/**
+ * Wraps interactions with the Google Cloud Run API
+ *
+ * @param region Region of the GCP resource.
+ * @param opts list of ClientOptions
+ * @returns CloudRun client
+ */
+export class CloudRun {
+  readonly methodOptions = {
+    userAgentDirectives: [
+      {
+        product: 'github-actions-deploy-cloudrun',
+        version: '0.1.0',
```

Comment:
I wonder if there's a way to pull this from the env. That way we don't have to remember to update this string with every release.

Reply:
Submitted a question / feature request https://github.com/actions/toolkit/issues/494